sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingQueryListenerBus.scala

```diff
@@ -35,12 +35,12 @@ import org.apache.spark.util.ListenerBus
  * and StreamingQueryManager. So this bus will dispatch events to registered listeners for only
  * those queries that were started in the associated SparkSession.
  */
-class StreamingQueryListenerBus(sparkListenerBus: LiveListenerBus)
+class StreamingQueryListenerBus(sparkListenerBus: Option[LiveListenerBus])
```

Comment:
Make `sparkListenerBus `optional. When loaded by in History Server, spark use the `ReplayListenerBus` instead of `LiveListenerBus`. In live ui, streaming query post the event into `StreamingQueryListenerBus`, and the `StreamingQueryListenerBus` will post these events into `LiveListenerBus`. `StreamingQueryListenerBus` also subscribes to `LiveListenerBus` events thus getting back the posted event in a different thread. In history ui, `StreamingQueryListenerBus` will subscribes to the `ReplayListenerBus`, and process those events through `onOtherEvent()` directly.

Reply:
Nit: how about keep `live: Boolean` here to divide the live and history scenario explicitly?

Reply:
@uncleGen Could you update the comment for `StreamingQueryListenerBus`?